### DISCUSSION TITLE
Battle: rainbow is too slow

### DISCUSSION THREAD

**Discussion by fheroes2bugs** [ihhub/fheroes2#7830] [General]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

When playing at high speed, slow rainbow dramatically increases the duration of the battle.

https://github.com/ihhub/fheroes2/assets/145407420/358c8c33-e011-4781-8b20-65482f484808

### Describe the possible solution

Make rainbow speed to depend on chosen speed of the batte.
![2023-09-28_002423](https://github.com/ihhub/fheroes2/assets/145407420/ef1d4fc1-f5cc-4975-a801-7515f81f3945)

**Comment by oleg-derevenetz** [comment]
> Make rainbow speed to depend on chosen speed of the batte.

If you take a closer look, you will see that the rainbow animation speed already depends on the battle speed setting. But rainbow animation doesn't disappear until the "good luck" sound ends. As far as I understand, this was the consensus at the time when it was implemented because the other options looked worse.

**Reply by Branikolog** [reply]
@oleg-derevenetz , hello.
Indeed, the "good luck" sound is quite a long one and without waiting until it finishes, battle sounds are turning into a cacophony. Current behaviour was implemented intentionally. However, for the higher speeds I think a slight change could be made allowing to proceed the battle a few moments before sound finishes. Like after 80-90% of the sound was played already (for 10 and 9 respectively). It will slightly interfere with all next sounds, but would make the perception battle speed more matching.

**Comment by ihhub** [comment]
I'm moving this to the discussion topic as we had chats about this in the past.

**Comment by fheroes2bugs** [comment]
So what are the pros of slow rainbow?
Leadership (double turn) is fast, but luck (rainbow) is slow, there is no logic in it.

**Comment by Branikolog** [comment]
> So what are the pros of slow rainbow? Leadership (double turn) is fast, but luck (rainbow) is slow, there is no logic in it.

Indeed. We need a consistent logic for both of these events. Either make morale event last longer, or don't wait until luck sound finishes to proceed other battle actions.
